rts/lra/jaxrs/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -431,6 +529,23 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
                         Response.Status.ACCEPTED.getStatusCode(),
                         Response.Status.OK.getStatusCode());
             }
+
+            /*
+             * report any failed steps (ie if progress contains any failures) to the caller.
+             * If either filter encountered a failure they may have completed partial actions and
+             * we need tell the caller which steps failed and which ones succeeded. We use a
+             * different warning code for each scenario:
+             */
+            if (progress != null) {
```

Comment:
Would not be clearer and faster not to concatenate the progress ops in case of succesful run (ie. the most probably much more often case than the failure)? 
Checking here for empy ops failure or similar, instead of checking the content of string of `failureMesssage`?

Reply:
Let's deal with performance enhancements when we have the primary functionality in place. There are many places throughout the code that need to be optimised.